### PR TITLE
CAESinkPipewire: const updates and improvements

### PR DESCRIPTION
**Pull request by lrusak**
This PR uses the new constexpr map class added in #21084 

This allows marking some maps as constexpr.

I've also added a commit to fix a shadow variable.

No functional changes are present.

### CODE REVIEW

File: xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp [hunk 1]
```diff
@@ -220,22 +220,24 @@ void CAESinkPipewire::EnumerateDevicesEx(AEDeviceInfoList& list, bool force)
   auto loop = pipewire->GetThreadLoop();
   loop->Lock();
 
-  CAEDeviceInfo device;
-  device.m_deviceType = AE_DEVTYPE_PCM;
-  device.m_deviceName = "Default";
-  device.m_displayName = "Default";
-  device.m_displayNameExtra = "Default Output Device (PIPEWIRE)";
-  device.m_wantsIECPassthrough = true;
+  CAEDeviceInfo defaultDevice;
```
Comment (notspiff) [inline xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp:223]: use brace initialization / initlist ?

**Review by notspiff — APPROVED** [review]
Just one small suggestion

File: xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp (hunk 1 above)
Comment (lrusak) [reply xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp:223]: do you mean
```
  CAEDeviceInfo defaultDevice = {};
```
or
```
  CAEDeviceInfo defaultDevice = {
    .m_deviceName = "Default",
    .m_displayName = "Default",
    .m_displayNameExtra = "Default Output Device (PIPEWIRE)",
    .m_deviceType = AE_DEVTYPE_PCM,
    .m_wantsIECPassthrough = true,
  };
```

Comment (notspiff) [reply xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp:223]: I mean the latter except the named init is c++20 so it would be without the names.

Comment (lrusak) [reply xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp:223]: I don't really see a benefit as it decreases readability and requires mixed ways of setting members anyway.
```
 CAEDeviceInfo defaultDevice = {
    "Default",
    "Default",
    "Default Output Device (PIPEWIRE)",
    AE_DEVTYPE_PCM,
  };

  defaultDevice.m_wantsIECPassthrough = true;

```

Comment (notspiff) [reply xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp:223]: Benefit is no need to call default ctors for strings followed by assignment. But never mind, it isnt perf critical code.

**Pull request merged**